wkcuber/tile_cubing.py

```diff
@@ -1,60 +1,233 @@
+def check_input_pattern(input_pattern: str) -> str:
+    x_match = re.search("{x+}", input_pattern)
+    y_match = re.search("{y+}", input_pattern)
+    z_match = re.search("{z+}", input_pattern)
 
+    if x_match is None or y_match is None or z_match is None:
+        raise ArgumentTypeError("{} is not a valid pattern".format(input_pattern))
 
-def parse_tile_file_name(filename):
-    m = CUBE_REGEX.search(filename)
-    if m is None:
-        return None
-    return (int(m.group(1)), int(m.group(2)), m.group(3))
+    return input_pattern
 
 
-def find_source_files(source_section_path):
-    all_source_files = [
-        path.join(source_section_path, s)
-        for s in find_files(
-            path.join(source_section_path, "**", "*"), image_reader.readers.keys()
-        )
-    ]
+def replace_coordinates(
+    pattern: str, coord_ids_with_replacement_info: Dict[str, Tuple[int, int]]
+) -> str:
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        coord = occurrence[1]
+        if coord in coord_ids_with_replacement_info:
+            number_of_digits = (
+                len(occurrence) - 2 - coord_ids_with_replacement_info[coord][1]
+            )
+            if number_of_digits > 1:
+                format_str = "0" + str(number_of_digits) + "d"
+            else:
+                format_str = "d"
+            pattern = pattern.replace(
+                occurrence,
+                format(coord_ids_with_replacement_info[coord][0], format_str),
+                1,
+            )
+    return pattern
+
+
+def replace_coordinates_with_glob_regex(pattern: str, coord_ids: Dict[str, int]) -> str:
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        coord = occurrence[1]
+        if coord in coord_ids:
+            number_of_digits = len(occurrence) - 2 - coord_ids[coord]
+            pattern = pattern.replace(occurrence, "[0-9]" * number_of_digits, 1)
+    return pattern
 
-    all_source_files.sort()
-    return all_source_files
 
+def get_digit_numbers_for_dimension(pattern):
+    x_number = 0
+    y_number = 0
+    z_number = 0
+    occurrences = re.findall("({x+}|{y+}|{z+})", pattern)
+    for occurrence in occurrences:
+        if occurrence[1] == "x":
+            x_number = max(x_number, len(occurrence) - 2)
+        if occurrence[1] == "y":
+            y_number = max(y_number, len(occurrence) - 2)
+        if occurrence[1] == "z":
+            z_number = max(z_number, len(occurrence) - 2)
+    return x_number, y_number, z_number
+
+
+def detect_interval_for_dimensions(
+    file_path_pattern: str,
+    x_decimal_length: int,
+    y_decimal_length: int,
+    z_decimal_length: int,
+) -> int:
+    x_min = None
+    y_min = None
+    z_min = None
+    x_max = 0
+    y_max = 0
+    z_max = 0
+    arbitrary_file = None
+    file_count = 0
+    for x in range(x_decimal_length):
```

Comment:
See my other comment regarding the padding. The three for-loops should be avoidable, right?